protocol/messenger_handler.go

```diff
@@ -712,13 +725,12 @@ func (m *Messenger) HandleAcceptContactRequest(state *ReceivedMessageState, mess
 		return nil
 	}
 
-	// TODO: Handle missing contact request message
-	request, err := m.persistence.MessageByID(message.Id)
+	request, err := m.persistence.MessageByID(defaultContactRequestID(message.Id))
```

Comment:
This does not look right actually, are you sure it's what you want to do?

`defaultContactRequestID` is generally calculated over the contact.id, not the contact request id

Reply:
`message.Id` basically contains a contact ID at this point (does it need to be renamed in this case? or should the ID property actually be a request ID instead?

Reply:
(from my perspective, it's better and easier to be understood when `AcceptContactRequest` contains `ContactID` instead of just `ID`)

Reply:
mmh, I don't think it contains a `contact.ID`.

There are 2 kinds of contact requests:
1) Explicitly sent by the user, this one has a contact request id == to the message id sent
2) Default contact requests, these were added for backward compatibility / cases where the user's device data is lost. Those have an id derived by the contact.id (`defaultContactRequestID(contact.id))`.

But when you accept it, as far as I can tell, you always use the ID you received, and I don't believe we ever derive the id from `message.Id`, so I'd expect the line you added:

```
m.persistence.MessageByID(defaultContactRequestID(message.Id))
```
To never return any message, if I am not mistaking

Reply:
You're a bit mistaken — the `protobuf.AcceptContactRequest` was created with contactID. I changed it in the last commit though to make it a bit more clear.

Reply:
are you sure?

https://github.com/status-im/status-go/blob/5d80c5cc427cd5979a60122bd725e1380a5cfa55/protocol/messenger_contacts.go#L240

https://github.com/status-im/status-go/blob/5d80c5cc427cd5979a60122bd725e1380a5cfa55/protocol/messenger_contacts.go#L909

both cases where it's create it seems to be using the id of the contact request (which makes sense), where did you see that it was using the contact id?

Reply:
But it all has been changed in this PR — now it sends back the current contact ID just because that's what actually needs to be used in order to update the local pending notification request for the user that requested the contact to be added.
It does work perfectly.

Reply:
did you revert the change?
it would have broken compatibility between clients btw, so wasn't quite working perfectly ;)

Reply:
yes, it's now `m.persistence.MessageByID(message.Id)` as it was before